From 20bb4af6d97719f250a0424e55b7342601a2b7ef Mon Sep 17 00:00:00 2001
From: Fam Zheng <famz@redhat.com>
Date: Mon, 14 Apr 2014 02:05:19 +0200
Subject: [PATCH 09/12] iscsi: Remember to set ret for iscsi_open in error case

RH-Author: Fam Zheng <famz@redhat.com>
Message-id: <1397441119-8563-1-git-send-email-famz@redhat.com>
Patchwork-id: 58457
O-Subject: [RHEL-7 0day qemu-kvm PATCH 9/7] iscsi: Remember to set ret for iscsi_open in error case
Bugzilla: 1083413
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>

Signed-off-by: Fam Zheng <famz@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit cd82b6fb4daf87f17742f7971ba6cb90d23e6703)
Signed-off-by: Fam Zheng <famz@redhat.com>
---
 block/iscsi.c | 1 +
 1 file changed, 1 insertion(+)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 block/iscsi.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/block/iscsi.c b/block/iscsi.c
index ff5ca20..9fe3be8 100644
--- a/block/iscsi.c
+++ b/block/iscsi.c
@@ -1408,6 +1408,7 @@ static int iscsi_open(BlockDriverState *bs, QDict *options, int flags,
     iscsi_readcapacity_sync(iscsilun, &local_err);
     if (local_err != NULL) {
         error_propagate(errp, local_err);
+        ret = -EINVAL;
         goto out;
     }
     bs->total_sectors = sector_lun2qemu(iscsilun->num_blocks, iscsilun);
-- 
1.7.1

